src/llama-hparams.h

```diff
@@ -229,7 +229,10 @@ struct llama_hparams {
     // number of layers for which has_kv() returns true
     uint32_t n_layer_kv() const;
 
-    bool is_masked_swa(llama_pos p0, llama_pos p1) const;
+    // note that this function uses different SWA parameters from those in the hparams
+    // TODO: think of a better place for this function
+    // TODO: pack the SWA params in a struct?
+    static bool is_masked_swa(uint32_t n_swa, llama_swa_type swa_type, llama_pos p0, llama_pos p1);
```

Comment:
Changed this to a static function.

Maybe it should become a member like this:

```suggestion
   bool is_masked_swa(uint32_t il, llama_pos p0, llama_pos p1) const;
```

But let's refactor this after the `master` stabilized.